black.py

```diff
@@ -1364,27 +1346,83 @@ def preceding_leaf(node: Optional[LN]) -> Optional[Leaf]:
     return None
 
 
-def is_delimiter(leaf: Leaf) -> int:
-    """Return the priority of the `leaf` delimiter. Return 0 if not delimiter.
+def is_split_after_delimiter(leaf: Leaf, previous: Leaf = None) -> int:
+    """Returns the priority of the `leaf` delimiter, given that it is a delimiter
+    that would cause a line break after the delimiter. Returns 0 if not a
+    delimiter in this case.
 
     Higher numbers are higher priority.
     """
     if leaf.type == token.COMMA:
         return COMMA_PRIORITY
 
-    if leaf.type in COMPARATORS:
-        return COMPARATOR_PRIORITY
+    if (
+        leaf.type in KWARGS
+        and leaf.parent
+        and leaf.parent.type in {syms.argument, syms.typedargslist}
+    ):
+        return KWARGS_PRIORITY
+
+    return 0
+
 
+def is_split_before_delimiter(leaf: Leaf, previous: Leaf = None) -> int:
+    """Returns the priority of the `leaf` delimiter, given that it is a delimiter
+    that would cause a line break after the delimiter. Returns 0 if not a
+    delimiter in this case.
+
+    Higher numbers are higher priority.
+    """
     if (
         leaf.type in MATH_OPERATORS
         and leaf.parent
         and leaf.parent.type not in {syms.factor, syms.star_expr}
     ):
         return MATH_PRIORITY
 
+    if leaf.type in COMPARATORS:
+        return COMPARATOR_PRIORITY
+
+    if (
+        leaf.type == token.STRING
+        and previous is not None
+        and previous.type == token.STRING
+    ):
+        return STRING_PRIORITY
+
+    if (
+        leaf.type == token.NAME
+        and leaf.value == 'for'
+        and leaf.parent
+        and leaf.parent.type in {syms.comp_for, syms.old_comp_for}
+    ):
+        return COMPREHENSION_PRIORITY
+
+    if (
+        leaf.type == token.NAME
+        and leaf.value == 'if'
+        and leaf.parent
+        and leaf.parent.type in {syms.comp_if, syms.old_comp_if}
+    ):
+        return COMPREHENSION_PRIORITY
+
+    if (leaf.type == token.NAME and leaf.value in LOGIC_OPERATORS and leaf.parent):
```

Comment:
nit: parens are spurious here. We will teach Black to do this transformation on its own 😏